src/models/test_step_hook_definition.js

```diff
@@ -0,0 +1,27 @@
+import PickleFilter from '../pickle_filter'
+import StepDefinition from './step_definition'
+
+export default class TestStepHookDefinition extends StepDefinition {
+  constructor(data) {
+    super(data)
+    this.pickleFilter = new PickleFilter({
+      tagExpression: this.options.tags,
+    })
+  }
+
+  appliesToTestCase({ pickle, uri }) {
+    return this.pickleFilter.matches({ pickle, uri })
+  }
+
+  getInvalidCodeLengthMessage() {
+    return this.buildInvalidCodeLengthMessage('0 or 1', '2')
+  }
+
+  getInvocationParameters({ hookParameter }) {
+    return [hookParameter]
+  }
```

Comment:
I don't think these hooks need invocation parameters (like what is passed to before/after hooks)